### PR TITLE
Add Circle `collidecircle()`

### PR DESCRIPTION
**Pull request by itzpr3d4t0r**
This PR adds the `collidecircle` function to the Circle class.

## Credits

**Geometry Project:**
For code, docs and tests:
@novialriptide @Emc2356 @itzpr3d4t0r @ScriptLineStudios @avaxar @Matiiss @newpaxonian @maqa41 @blankRiot96
Also thanks to @Starbuck5 for kickstarting the idea and the occasional help!

**Functionality added in this PR**
- `circle.c`
https://github.com/pygame-community/pygame-geometry/blame/main/src_c/circle.c
Credits to @Emc2356 @itzpr3d4t0r @novialriptide
- `geometry.pyi`
https://github.com/pygame-community/pygame-geometry/blame/main/geometry.pyi
Credits to @Emc2356 @itzpr3d4t0r @novialriptide @ScriptLineStudios @avaxar 
- `geometry_test.py`
https://github.com/pygame-community/pygame-geometry/blame/main/test/test_circle.py
Credits to @itzpr3d4t0r @novialriptide
- `geometry.rst`
https://github.com/pygame-community/pygame-geometry/blame/main/docs/circle.rst
Credits to @itzpr3d4t0r
- `collisions.c`
Credits to @Emc2356 @itzpr3d4t0r @novialriptide @ScriptLineStudios @avaxar 
https://github.com/pygame-community/pygame-geometry/blame/main/src_c/collisions.c
- `collisions.h`
@Emc2356 @itzpr3d4t0r @novialriptide
https://github.com/pygame-community/pygame-geometry/blame/main/src_c/include/collisions.h
- `geometry.h`
https://github.com/pygame-community/pygame-geometry/blame/main/src_c/include/geometry.h
Credits to @Emc2356 @itzpr3d4t0r @maqa41

### CODE REVIEW

File: buildconfig/stubs/pygame/geometry.pyi [hunk 1]
```diff
@@ -16,12 +35,16 @@ class Circle:
     @overload
     def __init__(self, pos: Sequence[float], r: float) -> None: ...
     @overload
-    def __init__(self, circle: Circle) -> None: ...
+    def __init__(self, circle: _CircleValue) -> None: ...
     @overload
     def __init__(self, obj_with_circle_attr) -> None: ...
```
Comment (ankith26) [inline buildconfig/stubs/pygame/geometry.pyi:40]: this line can be removed

File: buildconfig/stubs/pygame/geometry.pyi [hunk 2]
```diff
@@ -1,10 +1,29 @@
...
+from pygame import Vector3
 from pygame._common import Coordinate
 
+_CanBeCircle = Union[
+    Circle,
+    Tuple[float, float, float],
+    Tuple[Tuple[float, float], float],
+    Sequence[float],
+    Vector3,
+]
```
Comment (ankith26) [inline buildconfig/stubs/pygame/geometry.pyi:19]:
```suggestion
_CanBeCircle = Union[
    Circle,
    Tuple[Coordinate, float],
    Sequence[float]
]
```
Simplification ideas:
`Sequence[float]` already covers `Vector3` and `Tuple[float, float, float]`. And `Coordinate` is better than `Tuple[float, float]`

**Comment by ankith26** [comment]
There is a merge conflict here that needs to be fixed

Also I see you credited maqa41 in the initial comment, but did not add them as a co-author?

**Review by MyreMylar — APPROVED** [review]
LGTM 👍 

Works in local testing.

File: buildconfig/stubs/pygame/geometry.pyi [hunk 3]
```diff
@@ -49,12 +62,14 @@ class Circle:
...
-    def __init__(self, obj_with_circle_attr) -> None: ...
+    def __init__(self, circle: _CircleValue) -> None: ...
     @overload
     def collidepoint(self, x: float, y: float) -> bool: ...
     @overload
     def collidepoint(self, point: Coordinate) -> bool: ...
+    @overload
+    def collidecircle(self, circle: _CircleValue) -> bool: ...
+    @overload
+    def collidecircle(self, x: float, y: float, r: float) -> bool: ...
```
Comment (ankith26) [inline buildconfig/stubs/pygame/geometry.pyi:73]: Needs 
```py
    @overload
    def collidecircle(self, xy: Coordinate, r: float) -> bool: ...
```
too :)

**Review by ankith26 — APPROVED** [review]
LGTM (apart from one review), thanks to all involved for the PR :tada:

**Pull request merged**